refactor auto-update module

zeroxbt opened the pull request.

zeroxbt committed:
refactor autoupdate-module

zeroxbt committed:
add symlinks for safer updates

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
fix paths to files and folders

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
fix symlink creation

zeroxbt committed:
fix paths to symlink and folders

zeroxbt committed:
improve npm message logging

zeroxbt committed:
bum version for testing purposes

zeroxbt committed:
remove old version only if not named ot-node

zeroxbt committed:
bump version for testing purposes

zeroxbt committed:
modify comments

zeroxbt committed:
change version back to current (1.35)

djordjekovac committed:
Added folder structure initial migration

djordjekovac committed:
Run migrations only if testnet or mainnet env

djordjekovac committed:
Logging moved to base migration

zeroxbt committed:
refactor autoupdater

NZT48 commented on config/config.json:
```diff
@@ -46,12 +49,15 @@
+    "appDataPath": "data",
     "blockchain": [
       {
         "blockchainTitle": "ganache",
```
I think that we should now update all networkIDs to be ganache:testnet instead of ganache::testnet. cc: @kotlarmilos

kotlarmilos replied:
yes, but let's introduce it with the appropriate changes in the next week

djordjekovac committed:
Updated http module interface,
updated autoupdater module interface,
created base module interface,
updated initialization of all modules

vterzic commented on src/controller/http-api-router.js:
```diff
@@ -23,44 +23,44 @@ class HttpApiRouter {
     async initializeListeners() {
         // POST REQUESTS
-        this.httpClientModule.post(HTTP_API_ROUTES.PUBLISH, (req, res) => {
+        this.httpClientModuleInterface.post(HTTP_API_ROUTES.PUBLISH, (req, res) => {
             this.publishController.handleHttpApiPublishRequest(req, res);
         });
```
Is there a particular reason for wrapping route names inside of HTTP_API_ROUTES constant?

In my opinion this causes router obfuscation, since we can't see route name directly in the file and therefore can't search file by route name.

vterzic commented on src/controller/http-api-router.js:
```diff
@@ -23,44 +23,44 @@ class HttpApiRouter {
         });
 
-        this.httpClientModule.post(HTTP_API_ROUTES.PROVISION, (req, res) => {
+        this.httpClientModuleInterface.post(HTTP_API_ROUTES.PROVISION, (req, res) => {
             this.publishController.handleHttpApiProvisionRequest(req, res);
         });
```
Please consider removing unnecessary wrapper function for controller handler.

Instead of 
```
this.httpClientModuleInterface.post(HTTP_API_ROUTES.PROVISION, (req, res) => {
            this.publishController.handleHttpApiProvisionRequest(req, res);
});
```
we can just have
```
this.httpClientModuleInterface.post(HTTP_API_ROUTES.PROVISION, this.publishController.handleHttpApiProvisionRequest);
```

djordjekovac committed:
Configuration for autoupdater updated

vterzic commented on modules/migration/base-migration.js:
```diff
@@ -0,0 +1,61 @@
+
+class BaseMigration {
+    constructor(migrationName, logger, config) {
+        if (!migrationName || migrationName === '') {
```
Please consider removing || migrationName === '' since it will be covered within !migrationName check

vterzic commented on src/modules/auto-updater/implementation/ot-auto-updater.js:
```diff
@@ -0,0 +1,233 @@
+
+const REPOSITORY_URL = 'https://github.com/OriginTrail/ot-node';
+
+class OTAutoUpdater {
```
Please consider adding js docs for this class

djordjekovac committed:
Updated naming, interface to manager

djordjekovac committed:
OT-node udpate procedure and file system updated

djordjekovac committed:
Updated dependency installation steps

djordjekovac committed:
Added log if node is running the latest version

djordjekovac committed:
Node is now updating using zip archive from github

djordjekovac committed:
Added current symlink when installing new update

djordjekovac committed:
Updated current symlink in migration script

zeroxbt committed:
update installer to support new folder structure

zeroxbt committed:
Merge branch 'v6/refactor/auto-update-module' of https://github.com/OriginTrail/ot-node into v6/refactor/auto-update-module

djordjekovac committed:
Resolved issues with publish process

djordjekovac committed:
Merge branch 'v6/refactor/prepare-code-structure-for-updated-protocol' into v6/refactor/auto-update-module

# Conflicts:
#	modules/service/file-service.js
#	src/controller/v1/publish-controller.js

zeroxbt committed:
update noderc filepath

zeroxbt committed:
remove nodejs setup folder

zeroxbt committed:
add update file path

zeroxbt committed:
add absolute path to .env file

zeroxbt committed:
add migrations folder path to file service

zeroxbt committed:
update dependency injection paths to be absolute

zeroxbt committed:
remove unnecessary config declarations

zeroxbt committed:
update rc file path

zeroxbt committed:
add correct app name for rc

zeroxbt committed:
update ot-node download link

djordjekovac committed:
Version bump

zeroxbt committed:
update path for db migrations config

zeroxbt committed:
Merge branch 'v6/refactor/auto-update-module' of https://github.com/OriginTrail/ot-node into v6/refactor/auto-update-module

djordjekovac committed:
Updated npm install command

djordjekovac committed:
version bump

zeroxbt committed:
update paths

zeroxbt committed:
Merge branch 'v6/refactor/auto-update-module' of https://github.com/OriginTrail/ot-node into v6/refactor/auto-update-module

zeroxbt committed:
aupdate current directory

zeroxbt committed:
minor fix to sequelize command

zeroxbt committed:
revert changes and bump version

djordjekovac committed:
Updated otnode.service

djordjekovac committed:
Updated path to configuration for network module initialization

djordjekovac committed:
version bump

zeroxbt committed:
remove copy configuration file

zeroxbt committed:
Merge branch 'v6/refactor/auto-update-module' of https://github.com/OriginTrail/ot-node into v6/refactor/auto-update-module
